src/fshtypes/ValueSet.ts

```diff
@@ -0,0 +1,19 @@
+import { FshEntity } from './FshEntity';
+import { FshCode } from './FshCode';
+import { ValueSetTerm } from './ValueSetTerm';
+
+export class ValueSet extends FshEntity {
+  id: string;
+  description?: string;
+  codes: ValueSetTerm[];
+  includesDescendants: FshCode[];
+  excludesDescendants: FshCode[];
```

Comment:
I'm not sure if these are the only two ways that we can apply some sort of filter to a ValueSet. For example, I think something like this `* include concept is-a RACE#2133-7` could be done, and I'm not sure how that would be handled. There are a set of operators http://hl7.org/fhir/valueset-filter-operator.html, and I'm not sure if we have decided what is and isn't allowed. If you did already discuss this and decided this is what we need for now, sorry!

Reply:
You are correct that there are additional operators that could be applied. I was basing this implementation off https://github.com/HL7/fhir-shorthand/wiki/3.3-Value-Sets, which only lists `include descendant-of` and `exclude descendant-of` and mentions that it borrows a subset of the available operators.

Reply:
I haven't had a chance to review yet myself, but I saw these comments come by.  The intent is to support _all_ of the filter operators.  That being the case, I don't know if it's better to explicitly put each one in the grammar or if it might be better just to have a generic slot for the filter in the grammar that is then validated against the supported list in the importer.  If/when we eventually add support for DSTU2, it has a different set (well, a subset) of filter operators.

Reply:
It's fine to have them explicitly in the grammar, but a few things to note about that:
* although you find the error sooner, the error message may be harder for the user to understand (since it will be an ANTLR message complaining about not being able to find a matching parserule)
* if I recall correctly, ANTLR tries to do as exact a match as possible on lexer rules when creating the token stream; so any time you define something as a specific lexer token, that means that this is the token that will be used _wherever_ if finds that word.  For example, since `exists` is one of the filter ops, if you create a lexer rule `KW_EXISTS` for it, then if there is a FHIR property called exists and someone uses it in a definition:
  ```
  * exists = true
  ```
  then the lexer will make that a `KW_EXISTS` token even though the parse rule for that part of the grammar might be expecting a `SEQUENCE` token.  For this set of keywords, it's probably not a problem, but it's always something to think about when deciding whether to make a new lexer token or not.

As for where the filter ops can be used, yes -- they can be in the `include` or `exclude` rules.

Reply:
> ANTLR tries to do as exact a match as possible on lexer rules
I can't remember if that's right or if it's that ANTLR uses the *first* lexer rule that matches (going top-down).  Either way, the result would actually be the same because in order to get `KW_EXISTS` at all, it would have to be higher than `SEQUENCE`.